packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -1868,6 +1870,200 @@ function commitPassiveLifeCycles(finishedWork: Fiber): void {
   }
 }
 
+function commitDoubleInvokeEffectsInDEV(
+  fiber: Fiber,
+  hasPassiveEffects: boolean,
+) {
```

Comment:
Should we wrap the bodies of each of these newly added functions in `enableDoubleInvokingEffects` (or add an early return if `enableDoubleInvokingEffects` is false) like we do elsewhere? Just to guard against them possibly leaking into the build even if they're disabled?